File: idakonyvtar/src/main/java/eu/eyan/idakonyvtar/util/SpecialCharacterHelper.java
package eu.eyan.idakonyvtar.util;

import java.text.Normalizer;
import java.text.Normalizer.Form;
import java.util.regex.Pattern;

public class SpecialCharacterHelper
{

    public static Pattern filterPattern(String filterText)
    {
        return Pattern.compile(SpecialCharacterHelper.withoutSpecChars(filterText), Pattern.CASE_INSENSITIVE);
    }

    public static String withoutSpecChars(String text)
    {
        return text == null ? null
                : Normalizer.normalize(text, Form.NFD)
                        .replaceAll("\\p{InCombiningDiacriticalMarks}+", "");
    }

}

File: idakonyvtar/src/main/scala/eu/eyan/idakonyvtar/util/SpecialCharacterHelper.scala
package eu.eyan.idakonyvtar.util;

import java.text.Normalizer;
import java.text.Normalizer.Form;
import java.util.regex.Pattern;

object SpecialCharacterHelper {

  def filterPattern(filterText: String) = Pattern.compile(withoutSpecChars(filterText), Pattern.CASE_INSENSITIVE)

  def withoutSpecChars(text: String) = if (text == null) null else
    Normalizer.normalize(text, Form.NFD)
      .replaceAll("\\p{InCombiningDiacriticalMarks}+", "")
}